cmake/Utility.cmake

```diff
@@ -52,13 +52,21 @@ function(complex_enable_warnings)
         /we4458 # C4458: declaration of 'identifier' hides class member
         /we4459 # C4459: declaration of 'identifier' hides global declaration
     )
-  else()
+  elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
```

Comment:
Since clang and gcc are mostly the same, we can move the check:
```cmake
if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
  set(SHADOW_WARNING "shadow")
else()
  set(SHADOW_WARNING "shadow-all")
endif()

...

-Werror=${SHADOW_WARNING}
```

Reply:
Done